Aaron - LA144.building_det_en

aarony2j opened the pull request:
LA144.building_det_en #396

bpbond commented on R/constants.R:
```diff
@@ -32,6 +32,7 @@ MODEL_YEARS <- c(BASE_YEARS, FUTURE_YEARS)
 SSP_FUTURE_YEARS <- c(2010, FUTURE_YEARS)
 GHG_CONTROL_READIN_YEAR <- 1975
 BASE_YEAR_IFA <- 2006
+CLIMATE_NORMAL_YEARS <- 1981:2000
```
What does this mean? Should this instead go in energy section as `energy.CLIMATE_NORMAL_YEARS`?

aarony2j replied:
I don't know, so I opened up an issue (#587). In the meantime, I'll do what you suggest.

bpbond commented:
Note however that oldnew tests are not passing.

d3y419 commented:
@sypnl : can you see this? I request you to be the science reviewer for this chunk.

aarony2j commented:
Regarding the old new tests. I'm not seeing any failures. Am I missing something?
```
module_helpers: ....................................................................................
oldnew: ..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
pipeline-helpers: ...........................
timeshift: .........................................................................................................................................
util: ........
xml: .SSSSSS...SS......S......S

Skipped ------------------------------------------------------------------------
1. bogus MI header causes error (@test_xml_conv_utils.R#13) - Skipping test as global option gcamdata.use_java is not TRUE

2. converting without adding data causes error (@test_xml_conv_utils.R#23) - Skipping test as global option gcamdata.use_java is not TRUE

3. converting bogus data causes error (@test_xml_conv_utils.R#33) - Skipping test as global option gcamdata.use_java is not TRUE

4. can convert single table (@test_xml_conv_utils.R#44) - Skipping test as global option gcamdata.use_java is not TRUE

5. can convert multiple table (@test_xml_conv_utils.R#64) - Skipping test as global option gcamdata.use_java is not TRUE

6. get warning for missing header (@test_xml_conv_utils.R#86) - Skipping test as global option gcamdata.use_java is not TRUE

7. LandNode rename works (@test_xml_conv_utils.R#140) - Skipping test as global option gcamdata.use_java is not TRUE

8. add_node_equiv_xml works (@test_xml_conv_utils.R#162) - Skipping test as global option gcamdata.use_java is not TRUE

9. add_logit_tables_xml works (@test_xml_conv_utils.R#212) - Skipping test as global option gcamdata.use_java is not TRUE

10. add_logit_tables_xml sets default logit (@test_xml_conv_utils.R#252) - Skipping test as global option gcamdata.use_java is not TRUE

Warnings -----------------------------------------------------------------------
1. parse_csv_header works (@test_fileutil.R#153) - there is no package called 'R.utils'

DONE ===========================================================================
```

bpbond commented:
@aarony2j Let's see if the Travis rebuilds pass.

aarony2j commented:
@bpbond Note that I changed to FLAG_SUM_TEST for L144.end_use_eff.  I couldn't find the error, so I changed it per the instructions in test_oldnew.R. Not sure if that note is applicable in my case. Also, I wonder why these errors are not triggered when I'm in Rstudio.

bpbond commented:
>Note that I changed to FLAG_SUM_TEST for L144.end_use_eff

Need more documentation to do this--i.e. statistics showing that the data match to very high precision.

Re Travis erroring - this is happening because the `L101.in_EJ_ctry_bld_Fi_Yh` isn't guaranteed to be available (and it isn't, on Travis). The code needs to check for this and default to a pre-built output as necessary. See for example the `LA118.hydro` chunk. I'm around all this week to consult/talk about this if useful, or happy to just put it in.

bpbond commented:
Once the correct flags are used, `L144.end_use_eff` is not significantly different from the old data system:
```
> summary(comb$diff)   # value - old.value
      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-6.217e-15 -2.220e-16  5.551e-17  2.911e-17  3.331e-16  5.107e-15 
```

All tests now pass on my machine.

sypnl commented:
Looks great overall! Two minor comments: 1) spell out RG3; 2) clarify RG3 multiplier, as two multipliers were used to account for efficiency improvement and regional differences.

bpbond commented:
Thanks! @aarony2j make final changes, update, and we'll get this merged.

bpbond commented:
🎉

aarony2j commented:
@bpbond  do I have to open another PR to address Sha's comments?

bpbond commented:
Ah crap, you pushed and I thought that was it! Sorry. Do whatever's easiest. Another PR is fine.

aarony2j commented:
Ah, I see. I updated the branch from the master and synced. My bad! I'll open another PR. Before approving, let me get Sha to approve first in case she wants other/further changes. Thanks!
